ragstack-e2e-tests/e2e_tests/conftest.py

```diff
@@ -53,34 +61,50 @@ def get_astra_ref() -> AstraRef:
     )
 
 
-def delete_all_astra_collections_with_client(raw_client: LibAstraDB):
-    """
-    Deletes all collections.
+DEFAULT_ASTRA_REF = get_astra_ref()
+DEFAULT_ASTRA_CLIENT = LibAstraDB(
+    api_endpoint=DEFAULT_ASTRA_REF.api_endpoint, token=DEFAULT_ASTRA_REF.token
+)
 
-    Current AstraDB has a limit of 5 collections, meaning orphaned collections
-    will cause subsequent tests to fail if the limit is reached.
-    """
-    collections = raw_client.get_collections().get("status").get("collections")
-    logging.info(f"Existing collections: {collections}")
-    for collection_info in collections:
-        logging.info(f"Deleting collection: {collection_info}")
-        raw_client.delete_collection(collection_info)
 
+def ensure_astra_env_clean():
+    collections = (
+        DEFAULT_ASTRA_CLIENT.get_collections().get("status").get("collections")
+    )
+    if len(collections) >= 3:
+        logging.info(
+            f"Awaiting for 2 collection slots to be available, currently there are {len(collections)} collections"
+        )
+        delete_all_astra_collections()
+        ensure_astra_env_clean()
```

Comment:
Should we add a backoff to here so as to not overwhelm data api with delete timeouts, or perhaps in the `delete_all_astra_collections()`?

Reply:
Actually curious about this logic -- if deletions are running asynchronously from the tests, isn't it possible that this `delete_all` will delete a collection that we're currently using in a test?

Reply:
you're right, it's better to start a test with 0 collection on the db. 
also, I added a simple sleep between the retries - it should be enough